### DISCUSSION TITLE
How can i make query state persistent while app is intercepting route?

### DISCUSSION THREAD

**Discussion by kenaku** [47ng/nuqs#583] [Q&A] [answered]
Hey guys! 
Awesome lib, heavily rely on it in our mid-size nextjs app.
 
I recently rewritten it with App router approach and used [intercepted route](https://nextjs.org/docs/app/building-your-application/routing/intercepting-routes) pattern. 
Works great in general but this two patterns (intercepted routes and query state) are conflicting now. 

For example I have a list:  `https://customuse.com/collections/roblox?orderBy=popularity&isRemix=true`, state is in url. 
When I click on any item in the list, url is changed to `/d/:id` and modal window is opened.
Problem is that query params are cleared at this point and list is re-rendered in background with default filters while modal is opened. 

For now I solved it with useFilters hook with state in context that now is the source of truth, and query params are used only for cold start. So it works properly on example above.

Idk if its even possible to address this issue on library level, but decided to ask for advice either way) 
Really like this awesome DX experience on relying only on nuqs, without any contexts and providers.

**Comment by franky47** [comment, marked as answer]
The URL is the source of truth, so if you want these filters to persist in the item URL, they need to be present there.

The [serializer](https://nuqs.47ng.com/docs/utilities#serializer-helper) can help you with the URL recomposition, as it takes a base path to apply the query state on top of.
